andela/king-kong-ah-backend
# #166272104 Password Reset

#### What does this PR do?
This PR implements password reset feature

#### Description of Task to be completed?
- write tests for recover email and reset password features
- implement recover email feature
- implement reset password feature
- create middleware for validation

#### How should this be manually tested?
- input "http://localhost:3000/api/v1/auth/recoverEmail" post route on postman
- supply the user's email
- click on send
- go to your email, click on the "recover password" button 
- the button takes you to the browser, copy the link and input in postman
- input a new password and confirm password
- click on send

#### Any background context you want to provide?
User must have node, postgres and postman installed

#### What are the relevant pivotal tracker stories?
[166272104](https://www.pivotaltracker.com/story/show/166272104)
